### PR TITLE
Fix bug: PhpMyAdmin\Core::getRealSize / #15926

### LINKED ISSUE #15926: PhpMyAdmin\Core::getRealSize('8000M') returns a float instead of an int
### Describe the bug

There were some serious issues after upgrading the PHP of server to PHP version 7.4.
I tried to fix some of them. But there are still problems.
I also downloaded the latest version of the script so it might be compatible with PHP 7.4.
But it seems that PHPMyAdmin is not yet compatible with PHP 7.4.

The old script has been working with php 7.4, but it always shows errors. But it worked anyway. But the errors were annoying.

Error:

```
Fatal error: Uncaught TypeError: Return value of PhpMyAdmin\Core::getRealSize() must be of the type int, float returned in /service/control/phpmyadmin/libraries/classes/Core.php:429 Stack trace: #0 /service/control/phpmyadmin/libraries/classes/Config.php(1301): PhpMyAdmin\Core::getRealSize('8000M') #1 /service/control/phpmyadmin/libraries/classes/Config.php(161): PhpMyAdmin\Config->checkUploadSize() #2 /service/control/phpmyadmin/libraries/classes/Config.php(139): PhpMyAdmin\Config->checkSystem() #3 [internal function]: PhpMyAdmin\Config->__construct('/site/aterd.com...') #4 /service/control/phpmyadmin/vendor/symfony/dependency-injection/ContainerBuilder.php(1144): ReflectionClass->newInstanceArgs(Array) #5 /service/control/phpmyadmin/vendor/symfony/dependency-injection/ContainerBuilder.php(618): Symfony\Component\DependencyInjection\ContainerBuilder->createService(Object(Symfony\Component\DependencyInjection\Definition), Array, true, 'config') #6 /site/aterd.com/sub-p in /service/control/phpmyadmin/libraries/classes/Core.php on line 429
```

### To Reproduce

1. Upload phpmyadmin script to server
2. Open phpmyadmin/ in browser at client systems

### Expected behavior

See panel

### Server configuration

- Operating system: FreeBSD 12.0-RELEASE
- Web server: nginx/1.16.1
- Database version: 10.4.6-MariaDB
- PHP version: 7.4.2
- phpMyAdmin version: Version 5.1.0-dev

### Client configuration

- Browser: All
- Operating system: Gnu/Linux

### LINKED ISSUE #15932: PhpMyAdmin\Core::getRealSize('8G') php error
On local enviroment using php.ini I increased memory size for php.
As result getting error:
PhpMyAdmin version is 5.0.1
```
PHP Fatal error:  Uncaught TypeError: Return value of PhpMyAdmin\Core::getRealSize() must be of the type integer, float returned in \libraries\classes\Core.php:438
Stack trace:
#0 \libraries\classes\Config.php(1323): PhpMyAdmin\Core::getRealSize('8G')
#1 \libraries\classes\Config.php(128): PhpMyAdmin\Config->checkUploadSize()
#2 \libraries\classes\Config.php(104): PhpMyAdmin\Config->checkSystem()
#3 [internal function]: PhpMyAdmin\Config->__construct('G:\\hosts\\lazik\\...')
#4 \vendor\symfony\dependency-injection\ContainerBuilder.php(1144): ReflectionClass->newInstanceArgs(Array)
#5 \vendor\symfony\dependency-injection\ContainerBuilder.php(618): Symfony\Component\DependencyInjection\ContainerBuilder->createService(Object(Symfony\Component\DependencyInjection\Definition), Array, true, 'config')
#6 \vendor\symfony\dependency-injection\ in \libraries\classes\Core.php on line 438


```

### LINKED ISSUE #15936: phpMyAdmin 5.0.1 errors out on a 32 bit system
When opening the login page, nothing appears. 

The error log shows the following error:

> PHP Fatal error:  Uncaught TypeError: Return value of PhpMyAdmin\\Core::getRealSize() must be of the type int, float returned in .../libraries/classes/Core.php

One of the tests for the class in question also fails:

```
$ vendor/bin/phpunit test/classes/CoreTest.php
PHPUnit 8.5.2 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.3.11-1~deb10u1
Configuration: /var/www/html/phpmyadmin/phpunit.xml.dist
Error:         No code coverage driver is available

.........................E.....................................  63 / 110 ( 57%)
...............................................                 110 / 110 (100%)

Time: 1.03 seconds, Memory: 10.00 MB

There was 1 error:

1) PhpMyAdmin\Tests\CoreTest::testGetRealSize with data set #4 ('12gb', 12884901888.0)
TypeError: Return value of PhpMyAdmin\Core::getRealSize() must be of the type int, float returned

/var/www/html/phpmyadmin/libraries/classes/Core.php:429
/var/www/html/phpmyadmin/test/classes/CoreTest.php:486

ERRORS!
Tests: 110, Assertions: 181, Errors: 1.
```

### To Reproduce

Installing version 5.0.1 on for example a raspberry pi using the standard 32bit raspian reproduces the problem.

### To Potentially Solve

To  make phpMyAdmin usable, removing the return specification from the method seems to work. This also makes that section of the tests pass. Not a real solution unfortunately.

libraries/classes/Core.php (411)
```
    public static function getRealSize($size = 0)
```

### Server configuration

- Operating system: Linux raspberrypi 4.19.97-v7l+ #1294 SMP Thu Jan 30 13:21:14 GMT 2020 armv7l GNU/Linux
- Web server: Apache/2.4.38 (Raspbian) 
- Database version: Server version: 10.3.17-MariaDB-0+deb10u1 - Raspbian 10 
- PHP version: PHP version: 7.3.11-1~deb10u1 
- phpMyAdmin version: 5.0.1

### PR DESCRIPTION
**Pull request by BaseMax**
Signed-off-by: Max Base <Max [+.@.+] Asrez {_|.DOT.|_}com>

### Description

Issue: https://github.com/phpmyadmin/phpmyadmin/issues/15926

> PhpMyAdmin\Core::getRealSize('8000M') returns a float instead of an int

it seems that PHPMyAdmin is not yet compatible with PHP 7.4.2.
My Pull request is for **phpMyAdmin v5.1.0-dev**, Not old versions.

Reffer to: https://github.com/phpmyadmin/phpmyadmin/pull/15927
Sorry for my mistake in previous PR.

Fixes #15926

Before submitting pull request, please review the following checklist:

- [x] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [x] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [x] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [x] Every commit has a descriptive commit message.
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [x] Any new functionality is covered by tests.

### CODE REVIEW

**Comment by BaseMax** [comment]
Wow, **DCO** did accept my PR now.

**Review by williamdes — APPROVED** [review]

**Comment by williamdes** [comment]
@BaseMax can you add a test case to https://github.com/phpmyadmin/phpmyadmin/blob/27fbfc70d84b7b3e0dc5f4cc21ad176d738df188/test/classes/CoreTest.php#L543?

This PR is okay, thank you for fixing the signoff line

**Comment by williamdes** [comment]
If your real name "Seyyed Ali Mohammadiye" https://github.com/Asrez/Website/blob/9a643bc721e01f2c5734eb81f9c3dde6a1b84116/index.html#L7 ?
No insult, I am just afraid you are using a pseudonym

**Review by williamdes — CHANGES_REQUESTED** [review]
https://scholar.google.com/citations?user=0W-jUF0AAAAJ&hl=en

File: test/classes/CoreTest.php [hunk 1]
```diff
@@ -567,6 +567,14 @@ public function providerTestGetRealSize()
                 '1024',
                 1024,
             ],
+            [
+                '8000M',
+                8388608000
```
Comment (williamdes) [inline test/classes/CoreTest.php]:
```suggestion
                8388608000,
```

File: test/classes/CoreTest.php [hunk 2]
```diff
@@ -567,6 +567,14 @@ public function providerTestGetRealSize()
...
+            ],
+            [
+                '8G',
+                8589934592
```
Comment (williamdes) [inline test/classes/CoreTest.php]:
```suggestion
                8589934592,
```

**Comment by williamdes** [comment]
@BaseMax can you squash the commits and apply my suggestions ?
I am ready to merge

**Comment by williamdes** [comment]
Also why 8G != 8000M in your test data ?

**Comment by BaseMax** [comment]
> Also why 8G != 8000M in your test data ? @williamdes 

I did see both error in PHPmyAdmin.
So i did add both in unit tests. It's bad? I think we not have limitation in unit tests. They are not same.
If you want i will keep one.

File: test/classes/CoreTest.php (hunk 2 above)
Comment (BaseMax) [reply test/classes/CoreTest.php]: Added.

File: test/classes/CoreTest.php (hunk 1 above)
Comment (BaseMax) [reply test/classes/CoreTest.php]: Added.

**Comment by williamdes** [comment]
> > Also why 8G != 8000M in your test data ? @williamdes
> 
> Thanks you, I did see both error in PHPmyAdmin.
> So i did add both in unit tests. It's bad? I think we not have limitation in unit tests. They are not same.
> If you want i will keep one.
> 
> Cheers,

Oh sorry, you can add as much test cases as you want
I was just surprised that the result of 8G was different than 8000M
Or is it because it was storage units by 1024 and not 1000?

**Comment by williamdes** [comment]
Please squash your commits or give me access to your forked repo so I can fix it :)

**Comment by williamdes** [comment]
https://github.com/phpmyadmin/phpmyadmin/issues/15936#issuecomment-583844660

**Comment by BaseMax** [comment]
I did invite you, Also i did merge it. @williamdes

**Review by williamdes — APPROVED** [review]

**Comment by williamdes** [comment]
Thank you @BaseMax !
Sorry for the ping pongs to fix minor things, but it was worth it ;)

**Pull request merged**